lib/rust/ensogl/component/slider/src/model.rs

```diff
@@ -0,0 +1,168 @@
+//! Model for the slider component
+
+use ensogl_core::display::shape::*;
+use ensogl_core::prelude::*;
+
+use ensogl_core::application::Application;
+use ensogl_core::data::color;
+use ensogl_core::display;
+use ensogl_text as text;
+
+
+
+// ========================
+// === Model definition ===
+// ========================
+
+/// Slider model structure
+pub struct Model {
+    /// Background element
+    pub background:  background::View,
```

Comment:
You are using here things defined belo. In most cases, we try to define helper structs first, so the file can be read from top to bottom without jumping.